src/query/expression/src/utils/date_helper.rs

```diff
@@ -362,35 +365,157 @@ macro_rules! impl_interval_year_month {
     };
 }
 
-impl_interval_year_month!(AddYearsImpl, add_years_base);
-impl_interval_year_month!(AddMonthsImpl, add_months_base);
+impl_interval_year_month!(EvalYearsImpl, eval_years_base);
+impl_interval_year_month!(EvalMonthsImpl, eval_months_base);
+
+impl EvalYearsImpl {
+    pub fn eval_date_diff(date_start: i32, date_end: i32, tz: TzLUT) -> i32 {
+        let date_start = date_start.to_date(tz.tz);
+        let date_end = date_end.to_date(tz.tz);
+        date_end.year() - date_start.year()
+    }
+
+    pub fn eval_timestamp_diff(date_start: i64, date_end: i64, tz: TzLUT) -> i64 {
+        let date_start = date_start.to_timestamp(tz.tz);
+        let date_end = date_end.to_timestamp(tz.tz);
+        date_end.year() as i64 - date_start.year() as i64
+    }
+}
+
+pub struct EvalQuartersImpl;
+
+impl EvalQuartersImpl {
+    pub fn eval_date_diff(date_start: i32, date_end: i32, tz: TzLUT) -> i32 {
+        let diff_months = EvalMonthsImpl::eval_date_diff(date_start, date_end, tz);
+        diff_months.signum() * diff_months.abs().div_ceil(3)
```

Comment:
Now Databen has ToQuarter calc quartrter.

```rust
impl ToNumber<u8> for ToQuarter {
    fn to_number(dt: &DateTime<Tz>) -> u8 {
        (dt.month0() / 3 + 1) as u8
    }
}
```

https://github.com/databendlabs/databend/blob/a58d8b42aec4d38c2cffd40d2e6eb9ac2cf6bc90/src/query/expression/src/utils/date_helper.rs#L499

Directly calc diff and div 3 maybe cause some mistake.